compiler/damlc/lib/DA/Cli/Damlc/Test.hs

```diff
@@ -306,14 +306,14 @@ printTestCoverage ShowCoverage{getShowCoverage} allPackages results
           , printf "  %d defined" defined
           , printf "  %d (%5.1f%%) exercised" exercised (pctage exercised defined)
           ] ++ showCoverageReport printChoiceIdentifier "internal template choices never exercised" neverExercised
-        , let defined = countWhere (isLocal . fst) allImplementations
-              internal = countWhere (isLocal . fst) allImplementations
-              external = countWhere (not . isLocal . fst) allImplementations
+        , let localImplementations = M.filter (isLocal . fst) allImplementations
+              defined = M.size localImplementations
+              (internal, external) = M.partitionWithKey (\(ifaceId, _) _ -> isNothing (package ifaceId)) localImplementations
```

Comment:
Trying to see if I understand the change correctly. Before we were using local = internal, external = nonLocal, but apparently we should have only used local definitions anyway, and instead should check whether our local definition have a package definition, and if so, they are external, and if not, they are internal?

Reply:
almost. for context, remember this is the section for interface instances defined in this package (i.e. internal), and we want to classify them depending on where the _interface_ of the instance was defined. As you said, before we had `external = not . isLocal`, but using that on `allImplementations` gives us the interface instances **defined in other packages**. With @dylant-da's fix, we now look at the package of the _ifaceId_ of the interface and use that to classify the interface instances.